master/buildbot/changes/gitpoller.py

```diff
@@ -226,11 +225,8 @@ def _checkGitFeatures(self):
             if has_ssh_private_key:
                 raise EnvironmentError('SSH private keys require Git 2.3.0 or newer')
 
-    @defer.inlineCallbacks
     def activate(self):
         try:
-            self.lastRev = yield self.getState('lastRev', {})
```

Comment:
There can be only a single gitpoller which updates the `lastRev` state. Storing the newest value in `self` is perfectly acceptable way to avoid queries to the database.

Reply:
As it's one simple read query every `pollInterval`, I felt it didn't need the (albeit small) pain of maintaining the same state value in two places.

I could change this PR to simply:
- set `lastRev` to None in ctor and `reconfigService`
- in `poll`, if `lastRev` is None, load it from DB with `getState`.

I think it would solve the issue as well (need to confirm). Let me know if you prefer this solution.